packages/array-flatten/index.cjs

```diff
@@ -7,20 +7,32 @@ module.exports = flatten;
 
 function flattenHelper(arr, depth) {
   var stack = arr.slice();
-  var result = [];
+  var stackContainsArray = false;
 
-  while (stack.length) {
-    var item = stack.pop();
+  for (var i=0; i<stack.length; i++) {
+    if (Array.isArray(stack[i])) {
+      stackContainsArray = true;
+    }
+  }
```

Comment:
```suggestion
  var stackContainsArray = true;
```
I can always assume that stack contains array and the logic would still work but I was not 100% sure on if there exists a test case where it can break so I added another loop for it.

Let me know your thoughts on this.